x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -459,9 +462,41 @@ public void loadAuthorizedIndices(
         Map<String, IndexAbstraction> indicesLookup,
         ActionListener<Set<String>> listener
     ) {
+        throw new UnsupportedOperationException();
+    }
+
+    private final ResultDeduplicator<AuthorizedIndicesCacheKey, Set<String>> authorizedIndicesDeduplicator = new ResultDeduplicator<>();
+
+    @Override
+    public void loadAuthorizedIndices(
+        RequestInfo requestInfo,
+        AuthorizationInfo authorizationInfo,
+        Metadata metadata,
+        ActionListener<Set<String>> listener
+    ) {
+
         if (authorizationInfo instanceof RBACAuthorizationInfo) {
             final Role role = ((RBACAuthorizationInfo) authorizationInfo).getRole();
-            listener.onResponse(resolveAuthorizedIndicesFromRole(role, requestInfo, indicesLookup));
+            final TransportRequest request = requestInfo.getRequest();
+            final List<IndicesPermission> indicesPermissions;
+            if (role instanceof LimitedRole limitedRole) {
+                indicesPermissions = limitedRole.getIndicesPermissions();
+            } else {
+                indicesPermissions = List.of(role.indices());
+            }
```

Comment:
This is a temporary hack to get it working. The final code should do something different so that RBACEngine does not need deal with LimitedRole here.